### PR TITLE
[WIP] Reorganizing the bulk update buttons.

### LINKED ISSUE #87: Bulk Update - UI Changes
This is a simple rearranging of the current elements for clarity and space since it confuses some users that new questions open up below the bottom of the screen.  There are placeholders for new features which can be ignored. Image can be rev'ed to match current feature set.

bulk-update

### PR DESCRIPTION
**Pull request by tingulfsen**
There are no tests for this because the goal was to spend the minimum amount of time improving an interface that we already want to get rid of. Fixes #87

### CODE REVIEW

File: app/assets/stylesheets/argo/argo.sass [hunk 1]
```diff
@@ -189,3 +191,13 @@ button.close
...
+.bulk-buttons
+  margin-left: 5px
+  margin-right: 5px
+
+.bulk-buttons-row
+  margin-top: 8px
+  margin-bottom: 8px
+
+.bulk-btn-danger
+  background-color: $color-cardinal-red
```
Comment (mejackreed) [inline app/assets/stylesheets/argo/argo.sass]: Prefer `$brand-danger` here instead. Cardinal-red is a SU branding color, and shouldn't denote a dangerous item.

File: app/views/report/bulk.html.erb [hunk 2]
```diff
@@ -66,50 +66,77 @@ ul li{
...
-      <li><span class="bulk_button" class="update_buttons" onclick="$('#set_collection').show(400)">Set collection</span>
-      <li><span class="bulk_button" class="update_buttons" onclick="$('#add_collection').show(400)">Add collection</span>
-    </ul>
+  <div id="buttons" class="row bulk-buttons">
+      <div id="versioning_heading" class="row bulk-buttons-row">
+	  <strong>Options that require versioning</strong>
+      </div>
+      <div id="versioning_buttons" class="row bulk-buttons-row">
+	  <div id="versioning_buttons_col_1" class="col-md-2">
+	      <div id="col_1_row_1" class="row">
```
Comment (mejackreed) [inline app/views/report/bulk.html.erb]: Why were these given id's ? Are they being used by the JavaScript?

File: app/views/report/bulk.html.erb [hunk 3]
```diff
@@ -66,50 +66,77 @@ ul li{
...
+		  <span class="bulk_button btn-block" id="prepare" name="prepare" onclick="$('#open').show(400);">Prepare objects</span>
```
Comment (mejackreed) [inline app/views/report/bulk.html.erb]: Could these elements also be buttons? Then we could use the `btn` classes that bootstrap gives us.

File: app/views/report/bulk.html.erb [hunk 4]
```diff
@@ -66,50 +66,77 @@ ul li{
...
-      <label>Object Updates, you must prepare first!</label>
-      <li><span class="bulk_button" class="update_buttons" disabled id="show_source_id" name="show_source_id" onclick="$('#source_id').show(400);get_source_ids();">Source Id</span>
-      <li><span class="bulk_button" class="update_buttons" disabled onclick="$('#content_type').show(400)">Content type</span>
-      <li><span class="bulk_button" class="update_buttons" disabled onclick="$('#rights').show(400)">Object rights</span>
-      <li><span class="bulk_button" class="update_buttons" onclick="$('#set_collection').show(400)">Set collection</span>
-      <li><span class="bulk_button" class="update_buttons" onclick="$('#add_collection').show(400)">Add collection</span>
-    </ul>
+  <div id="buttons" class="row bulk-buttons">
+      <div id="versioning_heading" class="row bulk-buttons-row">
+	  <strong>Options that require versioning</strong>
```
Comment (mejackreed) [inline app/views/report/bulk.html.erb]: Seems like some of the whitespace here is a mix of tabs and spaces. I think 2 spaces is the what we use in most places.

File: app/assets/stylesheets/argo/argo.sass (hunk 1 above)
Comment (tingulfsen) [inline app/assets/stylesheets/argo/argo.sass]: Yeah good point, I will change this.

File: app/views/report/bulk.html.erb (hunk 2 above)
Comment (tingulfsen) [inline app/views/report/bulk.html.erb]: Actually I think I intended to write some row/column - specific CSS at one point, but didn't. I ended up leaving them in there. Are you saying it is better to remove the unused IDs?

Comment (mejackreed) [inline app/views/report/bulk.html.erb]: Yes. My opinion is that code not being used shouldn't be left in there.

File: app/views/report/bulk.html.erb (hunk 3 above)
Comment (mejackreed) [inline app/views/report/bulk.html.erb]: Hmm.. realizing that might be out of scope of what can be accomplished with this modification.

**Comment by mejackreed** [comment]
@tingulfsen One potential downside of doing non-column based rows is on varying screen resolutions you may get buttons to stack up in an undesired way:

screen shot 2015-12-10 at 10 47 20 am

**Comment by LynnMcRae** [comment]
One of the goals of reorganizing the page was to move the decision making up to the upper half of the screen, so when the action finally showed up, it was not off/below screen.  Stacking the buttons undoes this.

**Comment by mejackreed** [comment]
@LynnMcRae I think we might be talking about two different things here. Maybe huddle up quickly to discuss.

**Comment by tingulfsen** [comment]
I need to do some cleanup after this latest push,just looking for feedback on the HTML.

File: app/assets/stylesheets/argo/argo.sass [hunk 5]
```diff
@@ -131,8 +131,10 @@ table
 
 .bulk_button
   cursor: pointer
-  width: 100px
-  padding: 4px 6px
+  width: 160px
```
Comment (mejackreed) [inline app/assets/stylesheets/argo/argo.sass]: By using the Bootstrap `btn` and `btn-lg` classes you could probably get rid of these classes

File: app/assets/stylesheets/argo/argo.sass [hunk 6]
```diff
@@ -183,3 +185,13 @@ button.close
...
 
+.bulk-buttons
+  margin-left: 5px
+  margin-right: 5px
+
+.bulk-buttons-row
+  margin-top: 16px
+  margin-bottom: 16px
+
+.bulk-btn-danger
```
Comment (mejackreed) [inline app/assets/stylesheets/argo/argo.sass]: Since we are now using buttons here, we could remove this in favor of `btn-danger`

**Comment by mejackreed** [comment]
Looks good! Some minor comments on the classes, don't think that's a blocker. Though I do think the git history will need a clean up. Let's discuss on Monday.

**Comment by atz** [comment]
Grab me if you want some help squashing these commits.

**Comment by atz** [comment]
I fear you have rebased into crazy town.  Let me know if you want some help digging out or go ahead and close this PR if you plan to push a different clean branch.

**Comment by tingulfsen** [comment]
Closing, this pull request is superseded by #379

**Pull request closed without merging**